root/layout/components/BottomMenu.js

```diff
@@ -272,28 +275,31 @@ const DocumentationMenu = () => (
   </li>
 );
 
-type BottomMenuProps = {|
-  +currentLanguage: string,
-  +serverLanguages?: $ReadOnlyArray<ServerLanguageT>,
-|};
-
-const BottomMenu = ({
-  currentLanguage,
-  serverLanguages,
-}: BottomMenuProps) => (
+const BottomMenu = () => (
   <div className="bottom">
     <ul className="menu">
       <AboutMenu />
       <ProductsMenu />
       <SearchMenu />
-      {$c.user ? <EditingMenu /> : null}
+      <CatalystContext.Consumer>
+        {($c: CatalystContextT) => $c.user_exists ? <EditingMenu /> : null}
+      </CatalystContext.Consumer>
       <DocumentationMenu />
-      {(serverLanguages && serverLanguages.length > 1) ? (
-        <LanguageMenu
-          currentLanguage={currentLanguage}
-          serverLanguages={serverLanguages}
-        />
-      ) : null}
+      <CatalystContext.Consumer>
```

Comment:
Good question. Based on my reading of the docs and playing around with the code, having multiple consumers (of the same context) within the same `render()` function is equivalent to just having one consumer at the top-level of the `render()`, because the entire component has to re-render in either case (when the context changes). So I'll adjust the code accordingly.

Reply:
Ok, all components that need `$c` just use `withCatalystContext` now. Makes the diff smaller too.